aviary/subsystems/propulsion/test/test_propeller_performance.py

```diff
@@ -574,5 +575,95 @@ def test_tipspeed(self):
         assert_check_partials(partial_data, atol=5e-4, rtol=1e-4)
 
 
+class SquareRatioTest(unittest.TestCase):
+    """
+    Test the computation of square ratio with a maximum
+    """
+
+    def test_sqa_ratio_1(self):
+        prob = om.Problem()
+        prob.model.add_subsystem(
+            "group",
+            AreaSquareRatio(num_nodes=3, smooth_sqa=False),
+            promotes=["*"],
+        )
+        prob.setup(force_alloc_complex=True)
+        prob.set_val("DiamNac", val=2.8875, units='ft')
+        prob.set_val("DiamProp", val=10.0, units='ft')
+        prob.run_model()
+
+        sqa_ratio = prob.get_val("sqa_array", units='unitless')
+        assert_near_equal(sqa_ratio, [
+            0.08337656, 0.08337656, 0.08337656], tolerance=1e-5)
+
+        partial_data = prob.check_partials(out_stream=None, method="cs")
+        assert_check_partials(partial_data, atol=1e-12, rtol=1e-12)
+
+    def test_sqa_ratio_2(self):
+        prob = om.Problem()
+        prob.model.add_subsystem(
+            "group",
+            AreaSquareRatio(num_nodes=3, smooth_sqa=True),
+            promotes=["*"],
+        )
+        prob.setup(force_alloc_complex=True)
+        prob.set_val("DiamNac", val=2.8875, units='ft')
+        prob.set_val("DiamProp", val=10.0, units='ft')
+        prob.run_model()
+
+        sqa_ratio = prob.get_val("sqa_array", units='unitless')
+        assert_near_equal(sqa_ratio, [
+            0.08337656, 0.08337656, 0.08337656], tolerance=1e-5)
+
+        partial_data = prob.check_partials(out_stream=None, method="cs")
+        assert_check_partials(partial_data, atol=1e-12, rtol=1e-12)
+
+
+class AdvanceRatioTest(unittest.TestCase):
+    """
+    Test the computation of advanced ratio with a maximum
+    """
+
+    def test_zje_1(self):
+        prob = om.Problem()
+        prob.model.add_subsystem(
+            "group",
+            AdvanceRatio(num_nodes=3, smooth_zje=False),
+            promotes=["*"],
+        )
+        prob.setup(force_alloc_complex=True)
+        prob.set_val("vktas", val=[0.1, 125., 300.], units='knot')
+        prob.set_val("tipspd", val=[800., 800., 750.], units='ft/s')
+        prob.set_val("sqa_array", val=[0.0756, 0.0756, 0.0756], units='unitless')
+        prob.run_model()
+
+        equiv_adv_ratio = prob.get_val("equiv_adv_ratio", units='unitless')
+        assert_near_equal(equiv_adv_ratio, [
+            0.000650881807, 0.813602259, 2.08282178], tolerance=1e-5)
```

Comment:
for these two tests, could you add a fourth element that will trigger the limit?
Doesn't have to be "realistic" adding a 4th vktas that is 1000 would probably do it.

Reply:
Done adding unit tests that trigger the limits.